src/LanguageServer/Impl/CodeActions/MissingImportCodeActionProvider.cs

```diff
@@ -0,0 +1,635 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Analysis;
+using Microsoft.Python.Analysis.Analyzer;
+using Microsoft.Python.Analysis.Analyzer.Expressions;
+using Microsoft.Python.Analysis.Core.DependencyResolution;
+using Microsoft.Python.Analysis.Core.Interpreter;
+using Microsoft.Python.Analysis.Diagnostics;
+using Microsoft.Python.Analysis.Modules;
+using Microsoft.Python.Analysis.Types;
+using Microsoft.Python.Analysis.Values;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.Collections;
+using Microsoft.Python.Core.Text;
+using Microsoft.Python.LanguageServer.Diagnostics;
+using Microsoft.Python.LanguageServer.Indexing;
+using Microsoft.Python.LanguageServer.Protocol;
+using Microsoft.Python.Parsing.Ast;
+using Range = Microsoft.Python.Core.Text.Range;
+
+namespace Microsoft.Python.LanguageServer.CodeActions {
+    internal sealed class MissingImportCodeActionProvider : ICodeActionProvider {
+        private const int ModuleLoadTimeout = 10 * 1000; // 10 seconds
+
+        public static readonly ICodeActionProvider Instance = new MissingImportCodeActionProvider();
+
+        // right now, it is a static. in future, we might consider giving an option to users to customize this list
+        // also, right now, it is text based. so if module has same name, they will get same suggestion even if
+        // the module is not something user expected
+        private static readonly Dictionary<string, string> WellKnownAbbreviationMap = new Dictionary<string, string>() {
+            { "numpy", "np" },
+            { "pandas", "pd" }
```

Comment:
I think `tensorflow` is `tf`, `matplotlib.pyplot as plt`, `matplotlib as mpl`, `math as m`, `scipy.io as spio`, `scipy as sp` (https://docs.scipy.org/doc/scipy/reference/api.html, https://stackoverflow.com/questions/36014733/official-abbreviation-for-import-scipy-as-sp-sc)

Reply:
sure, I can do those.